dwave/system/samplers/clique.py

```diff
@@ -0,0 +1,186 @@
+                Any binary quadratic model with up to
+                :attr:`.largest_clique_size` variables. This BQM is embedded
+                using a dense clique embedding.
+
+            chain_strength (float, optional):
+                The (relative) chain strength to use in the embedding. By
+                default a chain strength of `1.5sqrt(N)` where `N` is the size
+                of the largest clique, as returned by
+                :attr:`.largest_clique_size`.
+
+            **kwargs:
+                Optional keyword arguments for the sampling method, specified
+                per solver in :attr:`.DWaveCliqueSampler.parameters`.
+                D-Wave System Documentation's
+                `solver guide <https://docs.dwavesys.com/docs/latest/doc_solver_ref.html>`_
+                describes the parameters and properties supported on the D-Wave
+                system. Note that `auto_scale` is not supported by this
+                sampler, because it scales the problem as part of the embedding
+                process.
+
+        """
+
+        # some arguments should not be overwritten
+        if 'auto_scale' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'auto_scale'")
+        if 'bias_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'bias_range'")
+        if 'quadratic_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'quadratic_range'")
+
+        # handle circular import. todo: fix
+        from dwave.system.composites.embedding import FixedEmbeddingComposite
+
+        # get the embedding
+        embedding = find_clique_embedding(bqm.variables, self.target_graph,
+                                          use_cache=True)
+
+        if chain_strength is None:
+            # chain length determines chain strength
+            if embedding:
+                chain_length = len(next(iter(embedding.values())))
+                chain_strength = 1.5 * math.sqrt(chain_length)
+            else:
+                chain_strength = 1  # doesn't matter
+
+            # scale chain strength by the problem scale
+            scalar = max(max(map(abs, bqm.linear.values()), default=0),
+                         max(map(abs, bqm.quadratic.values()), default=0))
+            chain_strength *= scalar
+
+        # # scaling only make sense in Ising space
+        original_bqm = bqm
+
+        if bqm.vartype is not dimod.SPIN:
+            bqm = bqm.change_vartype(dimod.SPIN, inplace=False)
+
+        sampler = FixedEmbeddingComposite(
+            dimod.ScaleComposite(self.child),
+            embedding)
+
+        sampleset = sampler.sample(bqm,
+                                   bias_range=self.qpu_linear_range,
+                                   quadratic_range=self.qpu_quadratic_range,
```

Comment:
I suppose that if you pass a `bias_range` and a `quadratic_range` at the same time it will rescale ALL the weights by the same magnitude so that linear and quadratic values are within their respective ranges, right?

In order words, this doesn't have the possibility of applying different rescalings to the linear and quadratic values. Correct?

The description is a bit confusing, it sounds like it would only normalize each type of bias independently instead of the whole Hamiltonian
```
            bias_range (number/pair):
                Value/range by which to normalize the all the biases, or if
                `quadratic_range` is provided, just the linear biases.

            quadratic_range (number/pair):
                Value/range by which to normalize the quadratic biases.```

Reply:
Yes, it picks a scalar that fits the whole problem within the desired range(s). Should fix the docs